pandas/_libs/parsers.pyx

```diff
@@ -445,10 +445,8 @@ cdef class TextReader:
         # suboptimal
         if usecols is not None:
             self.has_usecols = 1
-            if callable(usecols):
-                self.usecols = usecols
-            else:
-                self.usecols = set(usecols)
+            # GH20529, validate usecols at higher level.
```

Comment:
Let's expand this comment to this:

~~~
# gh-20558: we will validate and clean "usecols" before getting here.
~~~

Reply:
done.